### PR TITLE
[terra-navigation-side-menu] Added Drill-In view variant

### PR DESCRIPTION
**Pull request by sugan2416**
### Summary

**What was changed:**

Introduced Drill-In view variant under terra-navigation-side-menu.

**Why it was changed:**

Outline-View-Drill-In requirement by Mpages

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [x] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10382 
UXPLATFORM-10383
UXPLATFORM-10384
UXPLATFORM-10385
UXPLATFORM-10386
UXPLATFORM-10387
---

Thank you for contributing to Terra.
@cerner/terra

DrillIn View
image
image
image

### CODE REVIEW

File: packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx [hunk 1]
```diff
@@ -77,23 +86,34 @@ const propTypes = {
    * An optional toolbar to display below the side menu action header
    */
   toolbar: PropTypes.element,
+  /**
+   * Renders either Navigation Side Menu or Drill-IN
+   */
+  variant: PropTypes.oneOf(['nav-side-menu', 'drill-in']),
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx:92]: Use constants like in [button](https://github.com/cerner/terra-core/blob/3f404c33a4092f4e43dfcdbd86531727dd757b1b/packages/terra-button/src/Button.jsx#L107)

File: packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx [hunk 2]
```diff
@@ -306,17 +328,28 @@ class NavigationSideMenu extends Component {
...
+
+    if (this.props.variant !== 'drill-in' && key === 'empty-child-key') {
+      return null;
+    }
+
     return (
       <MenuItem
         id={item.id}
-        hasChevron={item.hasSubMenu || (item.childKeys && item.childKeys.length > 0)}
+        hasChevron={item.hasSubMenu || (item.childKeys && item.childKeys.length >= 0)}
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx:342]: Why '=' is added?

Comment (sugan2416) [reply packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx:342]: To add chevron to item/folder with no items/documents

File: packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx (hunk 1 above)
Comment (sugan2416) [reply packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx]: https://github.com/cerner/terra-framework/pull/2161/commits/066ee15428f028d455cea1e95f242f1cf9caef53

File: packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx (hunk 2 above)
Comment (saket2403) [reply packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx:339]: So is it supposed to be added for both the variants?

Comment (sugan2416) [reply packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx]: Reverted as no longer needed: https://github.com/cerner/terra-framework/pull/2161/commits/638d506e3d19e7a389d027232a52e61b0b0adfac

**Review by allanxavierm — APPROVED** [review]

**Comment by rahulkumar8cs** [comment]
+1

**Pull request merged**